src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -34,6 +34,12 @@ def live_only_execute(self, cli_ctx, command, expect_failure=False):
 
         return None
 
+    @property
+    def random_name(self):
+        # override random_name so that in play-back mode the name is deterministic.
+        # non-deterministic preparers have no recording of requests involved in creating the resource
+        return self.moniker
```

Comment:
If this is only supposed to apply to playback mode, then should it not be something like:
```
if self.is_live:
  return random_thing
else:
  return self.moniker
```
?